packages/ra-ui-materialui/src/detail/Create.tsx

```diff
@@ -59,19 +54,10 @@ export const Create = <RecordType extends RaRecord = any>(
     props: CreateProps<RecordType> & { children: ReactElement }
 ): ReactElement => {
     useCheckMinimumRequiredProps('Create', ['children'], props);
-    const controllerProps = useCreateController<RecordType>(props);
-    const body = (
-        <CreateContextProvider value={controllerProps}>
-            <CreateView {...props} {...controllerProps} />
-        </CreateContextProvider>
-    );
-    return props.resource ? (
-        // support resource override via props
-        <ResourceContextProvider value={props.resource}>
-            {body}
-        </ResourceContextProvider>
-    ) : (
-        body
+    return (
+        <CreateBase {...props}>
+            <CreateView {...props} />
```

Comment:
I'm not a fan of passing all props both to the controller and to the view. We should split them IMO (as we do in Show and List). 

The same goes for the Edit component.